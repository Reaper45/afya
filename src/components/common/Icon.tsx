import * as React from 'react'
import { ImgHTMLAttributes } from 'react'

import styled from 'lib/emotion'

const IconWrapper = styled("img")<{ active?: boolean }>`
  color: ${props =>
    props.active ? props.theme.colors.accent : props.theme.colors.primary};
`;

const Icon: React.FC<ImgHTMLAttributes<HTMLImageElement> & {
  active?: boolean;
}> = props => {
  return <IconWrapper {...props} />;
};

export default Icon;
