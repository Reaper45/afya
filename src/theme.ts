export interface ITheme {
  colors: {
    primary: string;
    dark: string;
    light: string;
    secondary: string;
    accent: string;
  };
}

export const theme: ITheme = {
  colors: {
    primary: "#5A67D8",
    dark: "#3C366B",
    light: "#A3BFFA",
    secondary: "#222222",
    accent: "A0AEC0"
  }
};
