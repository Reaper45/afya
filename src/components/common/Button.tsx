
import * as React from 'react'
import { ButtonHTMLAttributes } from 'react'

import styled from 'lib/emotion'
import Icon from './Icon';

const ButtonWrapper = styled("button")`
  outline: none;
  border: solid 1px ${props => props.theme.colors.grey};
  background: #fff;
  padding: .65rem 1rem;
  font-size: 0.8rem;
  color: ${props => props.theme.colors.secondary};
  border-radius: 3px;
  font-weight: 600;
  cursor: pointer;
  ::active {
    border-color: ${props => props.theme.colors.primary};
    color: ${props => props.theme.colors.primary};
  }
`;

const Button: React.FC<ButtonHTMLAttributes<HTMLButtonElement> & { icon?: string }> = ({ children, icon, ...props}) => {
  return (
    <ButtonWrapper {...props}>
      {children}
      {icon && <Icon src={icon} />}
    </ButtonWrapper>
  );
}

export default Button;
