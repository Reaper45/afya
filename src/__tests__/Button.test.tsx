import React, { SyntheticEvent } from "react";
import { shallow, mount } from "enzyme";
import { ThemeProvider } from "emotion-theming";

import { theme } from "theme";

import Button from "components/common/Button";
import Icon from "components/common/Icon";

import icon from "static/icons/icon-file.svg";

it("renders button correctly", () => {
  const component = shallow(<Button>Add Item</Button>);
  expect(component).toMatchSnapshot();
  expect(component.text()).toEqual("Add Item");
});

it("renders button with icon", () => {
  const component = shallow(<Button icon={icon}>Add Item</Button>);
  expect(component.containsMatchingElement(<Icon src={icon} />)).toBe(true);
});

it("handles click event", () => {
  const onClickHandler = jest.fn();
  const component = mount(
    <ThemeProvider theme={theme}>
      <Button onClick={onClickHandler}>Add Item</Button>
    </ThemeProvider>
  );

  component.find('button').at(0).simulate('click', { value: "clicked" })
  expect(onClickHandler).toHaveBeenCalled();
  expect(onClickHandler.mock.calls[0][0].value).toBe("clicked");
});