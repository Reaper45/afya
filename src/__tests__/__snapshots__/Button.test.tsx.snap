// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button correctly 1`] = `
<Styled(button)>
  Add Item
</Styled(button)>
`;
